### PR TITLE
Update testing

### PR DESCRIPTION
**Pull request by Simkern**
- preprocessor pass + macros added for TestExpm
- Test_utils module created for simplified initialization and data handling in tests
- `print_mat` subroutine added to utils for debugging purposes
- all tests cleaned and test utils deployed.

### CODE REVIEW

File: src/Utils.f90 [hunk 1]
```diff
@@ -135,6 +137,109 @@ subroutine zassert_shape(A, size, routine, matname)
...
+      character ( len = 10 ) iform
+      integer :: i, ihi, ilo, j, jhi, jlo, lmax, npline
+      logical :: integ
+      
+      write(*,*)
+      if (present(name)) then
+         write(*,*) 'Output matrix: ', trim(name)
+      endif
+      
+      ! Check if all entries are integral.
```
Comment (loiseaujc) [inline src/Utils.f90:158]: integral? Shouldn't the comment read `integer`?

**Review by loiseaujc — APPROVED** [review]

**Comment by loiseaujc** [comment]
Everything looks good. Only issue is that you sent a PR to `main` rather than `dev`. Let's take the habit of sending PR to `dev` first and will do the whole PR into `main`  afterward just to keep everything in sync.

**Comment by Simkern** [comment]
Sure, let's go to dev first, then to main.

File: src/Utils.f90 (hunk 1 above)
Comment (Simkern) [reply src/Utils.f90:158]: Yeah, I guess it's synonymous. I have no preference.

**Pull request closed without merging**